aptos-move/framework/aptos-framework/sources/simple_delegations.move

```diff
@@ -0,0 +1,848 @@
+        withdraw_delegation_events: EventHandle<WithdrawDelegationEvent>,
+    }
+
+    struct DelegateEvent has drop, store {
+        operator: address,
+        voter: address,
+        pool_address: address,
+        amount: u64,
+        commission_percentage: u64,
+    }
+
+    struct UpdateVoterEvent has drop, store {
+        operator: address,
+        pool_address: address,
+        old_voter: address,
+        new_voter: address,
+    }
+
+    struct RequestCommissionEvent has drop, store {
+        operator: address,
+        pool_address: address,
+        accumulated_rewards: u64,
+        commission_amount: u64,
+        previous_total_stake: u64,
+        updated_total_stake: u64,
+    }
+
+    struct WithdrawCommissionEvent has drop, store {
+        operator: address,
+        pool_address: address,
+        total_withdrawable_amount: u64,
+        amount_paid_to_current_operator: u64,
+    }
+
+    struct SwitchOperatorEvent has drop, store {
+        pool_address: address,
+        old_operator: address,
+        new_operator: address,
+        unpaid_commission_amount: u64,
+        updated_total_stake: u64,
+    }
+
+    struct EndDelegationEvent has drop, store {
+        pool_address: address,
+        operator: address,
+        unpaid_commission_amount: u64,
+    }
+
+    struct WithdrawDelegationEvent has drop, store {
+        operator: address,
+        pool_address: address,
+        total_withdrawable_amount: u64,
+        amount_withdrawn: u64,
+    }
+
+    public fun get_stake_pool_address(delegator: address, operator: address): address acquires Delegations {
+        let delegations = &borrow_global<Delegations>(delegator).delegations;
+        simple_map::borrow(delegations, &operator).pool_address
+    }
+
+    public fun get_last_recorded_total_stake(delegator: address, operator: address): u64 acquires Delegations {
+        let delegations = &borrow_global<Delegations>(delegator).delegations;
+        simple_map::borrow(delegations, &operator).total_stake
+    }
+
+    public fun get_commission_percentage(delegator: address, operator: address): u64 acquires Delegations {
+        let delegations = &borrow_global<Delegations>(delegator).delegations;
+        simple_map::borrow(delegations, &operator).commission_percentage
+    }
+
+    public fun is_delegation_active(delegator: address, operator: address): bool acquires Delegations {
+        let delegations = &borrow_global<Delegations>(delegator).delegations;
+        simple_map::borrow(delegations, &operator).is_active
+    }
+
+    /// Delegator can call this function to define a simple delegation contract with a specified operator.
+    ///
+    /// Can only delegate to a specific operator once. Afterward, delegator cannot update amount or commission.
+    public entry fun delegate(
+        delegator: &signer,
+        operator: address,
+        voter: address,
+        amount: u64,
+        commission_percentage: u64,
```

Comment:
`commission_percentage`, `voter`, and `operator` should be controlled by the manager of this not the delegator.

Reply:
Outdated.